test/dst/dst.go

```diff
@@ -122,16 +122,25 @@ func (d *DST) Run(r *rand.Rand, api api.API, aio aio.AIO, system *system.System,
 			metadata.Tags.Set("name", req.Kind.String())
 			metadata.Tags.Set("api", "dst")
 
+			if req == nil {
+				errs = append(errs, fmt.Errorf("request is nil"))
+				break
+			}
+
 			api.Enqueue(&bus.SQE[t_api.Request, t_api.Response]{
 				Metadata:   metadata,
 				Submission: req,
 				Callback: func(res *t_api.Response, err error) {
-					modelErr := model.Step(t, req, res, err)
-					if modelErr != nil {
-						errs = append(errs, modelErr)
+					// CQE Invariant: Assert that a callback is set after dequeuing
```

Comment:
for asserts we use util.Assert not this. and the invariant shouldn't be asserted here but in the AIO subsystem for production.